### PR TITLE
[v6r16] DynamicMonitoring Profiler

### PR DESCRIPTION
**Pull request by zmathe**
Added profiling utility used to retrieve dynamic information about a process. This is used by the dynamic monitoring system to get the information that will be stored into ElasticSearch.

### CODE REVIEW

File: Core/Utilities/Profiler.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
+"""
+Profiling class for updated information on process status
+"""
+
+__RCSID__ = "$Id$"
+
+import psutil
```
Comment (fstagni) [inline Core/Utilities/Profiler.py]: `psutil` is NOT in the externals: please make a PR adding it to https://github.com/DIRACGrid/Externals/blob/master/ServerPackages/dirac-make.py

And in this same PR you should add psutil to requirements.txt

File: Core/Utilities/Profiler.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    if result[ 'OK' ]:
+      data[ 'stats' ][ 'memoryUsage' ] = result[ 'Value' ]
+
+    result = self.numThreads()
+    if result[ 'OK' ]:
+      data[ 'stats' ][ 'threads' ] = result[ 'Value' ]
+
+    result = self.cpuUsage()
+    if result[ 'OK' ]:
+      data[ 'stats' ][ 'cpuUsage' ] = result[ 'Value' ]
```
Comment (fstagni) [inline Core/Utilities/Profiler.py]: Truly minor, but please change the spacing (e.g. `['stats']`)

File: Core/Utilities/Profiler.py (hunk 1 above)
Comment (zmathe) [inline Core/Utilities/Profiler.py]: Done

**Comment by zmathe** [comment]
recommended modifications are implemented

**Comment by fstagni** [comment]
You can use the new S_ERROR here, as well as in the other cases. If this new system is not defined, it's time to do it.

**Comment by fstagni** [comment]
The branch has conflicts

**Comment by zmathe** [comment]
yes. I will redo...

File: Core/Utilities/Profiler.py (hunk 2 above)
Comment (zmathe) [inline Core/Utilities/Profiler.py]: done

**Comment by zmathe** [comment]
recreated against v6r16 #3106

**Pull request closed without merging**